x-pack/plugins/observability/e2e/test_reporter.ts

```diff
@@ -0,0 +1,222 @@
+    if (!this.journeys.has(journey.name)) {
+      this.journeys.set(journey.name, []);
+    }
+    this.journeys.get(journey.name)?.push({ name: step.name, ...result });
+  }
+
+  async onJourneyEnd(journey: Journey, { error, start, end, status }: JourneyEndResult) {
+    const { failed, succeeded, skipped } = this.metrics;
+    const total = failed + succeeded + skipped;
+    if (total === 0 && error) {
+      this.write(renderError(error));
+    }
+    const message = `${symbols[status]} Took  (${renderDuration(end - start)} seconds)`;
+    this.write(message);
+
+    await fs.promises.mkdir('.journeys/failed_steps', { recursive: true });
+
+    await gatherScreenshots(join(CACHE_PATH, 'screenshots'), async (screenshot) => {
+      const { data, step } = screenshot;
+
+      if (status === 'failed') {
+        await (async () => {
+          await fs.promises.writeFile(join('.journeys/failed_steps/', `${step.name}.jpg`), data, {
+            encoding: 'base64',
+          });
+        })();
+      }
+    });
+  }
+
+  onEnd() {
+    const failedJourneys = Array.from(this.journeys.entries()).filter(([, steps]) =>
+      steps.some((step) => step.status === 'failed')
+    );
+
+    if (failedJourneys.length > 0) {
+      failedJourneys.forEach(([journeyName, steps]) => {
+        if (process.env.CI) {
+          const name = red(`Journey: ${journeyName} 🥵`);
+          this.write(`\n+++ ${name}`);
+          steps.forEach((stepResult) => {
+            const { status, end, start, error, name: stepName } = stepResult;
+            const message = `${symbols[status]}  Step: '${stepName}' ${status} (${renderDuration(
+              (end - start) * 1000
+            )} ms)`;
+            this.write(indent(message));
+            if (error) {
+              this.write(renderError(error));
+            }
+          });
+        }
+      });
+    }
+
+    const successfulJourneys = Array.from(this.journeys.entries()).filter(([, steps]) =>
+      steps.every((step) => step.status === 'succeeded')
+    );
+
+    successfulJourneys.forEach(([journeyName, steps]) => {
+      // fs.unlinkSync('.journeys/videos/' + journeyName + '.webm');
+    });
+
+    const { failed, succeeded, skipped } = this.metrics;
+    const total = failed + succeeded + skipped;
+
+    let message = '\n';
+    if (total === 0) {
+      message = 'No tests found!';
+      message += ` (${renderDuration(now())} ms) \n`;
+      this.write(message);
+      return;
+    }
+
+    message += succeeded > 0 ? green(` ${succeeded} passed`) : '';
+    message += failed > 0 ? red(` ${failed} failed`) : '';
+    message += skipped > 0 ? cyan(` ${skipped} skipped`) : '';
+    message += ` (${renderDuration(now() / 1000)} seconds) \n`;
+    this.write(message);
+  }
+
+  write(message: any) {
+    if (typeof message === 'object') {
+      message = JSON.stringify(message);
+    }
+    log(message + '\n');
+  }
+}
+
+const SEPARATOR = '\n';
+
+function indent(lines: string, tab = '   ') {
```

Comment:
All these can be exported from helpers - https://github.com/elastic/synthetics/blob/88a8fac88299666341f5d873aad3e356420f3204/src/helpers.ts#L54